service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java

```diff
@@ -130,6 +130,9 @@ public final void processCredentialsMessage(final Message<JsonObject> regMsg) {
         } catch (ClassCastException e) {
             log.debug("malformed request message");
             reply(regMsg, CredentialsResult.from(HTTP_BAD_REQUEST));
+        } catch (NullPointerException ne) {
```

Comment:
I have a hard time accepting a `NullPointerException` as an expected outcome. However, since the `CredentialsService` interface's methods explicitly declare `NullPointerException`s being thrown, I understand your intention. However, I think we should rather do explicit checks for `null` on the specific params within the cases of the `switch` statement and return a BAD_REQUEST replay on a case by case basis.

BTW I have the same problem with the `ClassCastException` being caught. I think we should instead do explicit checks in order to prevent the exception being caught.

WDYT?